src/olympia/lib/crypto/packaged.py

```diff
@@ -172,3 +179,21 @@ def is_signed(file_path):
         filenames = set()
     return set([u'META-INF/mozilla.rsa', u'META-INF/mozilla.sf',
                 u'META-INF/manifest.mf']).issubset(filenames)
+
+
+def get_signature_ou(pkcs7):
+    """
+    Extracts the ou out of a DER formatted, detached PKCS7
+    signature buffer
+    """
+    content = cms.ContentInfo.load(pkcs7)['content'].native
+
+    # Fetch the certificate stack that is the list of signers
+    # Since there should only be one in this use case, take the zeroth
+    # cert in the stack and return its OU
+    try:
+        cert = content['certificates'][0]['tbs_certificate']
```

Comment:
Note that looking for the first certificate only works because we happen to be putting the signer cert first. PKCS7 doesn't care about cert ordering, and we could very well move it around in the future.

The correct way to check this value is a bit more complicated:
1. parse `issuerAndSerialNumber` section of the `SignerInfo` sequence (there should be only one SignerInfo for add-ons, nss doesn't support multiples)
2. loop through each cert stored in the `certificates` section of the `SignedData` sequence
3. find the cert that matches the serial and issuer from the `issuerAndSerialNumber` section. That's the signer cert. Now verify that its first `ou` matches the string `Mozilla Extensions`.

That's quite a bit of code, and I don't know if you want to do that in AMO. The code you have here will work fine, but is not completely future-proof.